res/controllers/Numark-MixstreamPro-scripts.js

```diff
@@ -0,0 +1,1604 @@
+            engine.setValue("[Auxiliary1]", "master", 1)
+            engine.setValue("[Auxiliary1]", "pregain", 0.1)
+             
+
+            if (engine.getValue("[Channel1]", "play_indicator") === 1) {
+                engine.setValue("[Auxiliary1]", "orientation", 2)
+
+            } else
+            if (engine.getValue("[Channel2]", "play_indicator") === 1) {
+                engine.setValue("[Auxiliary1]", "orientation", 0)
+            }
+
+            MixstreamPro.WIFI = false
+        } else
+        if (MixstreamPro.WIFI === false) {
+            engine.setValue("[Auxiliary1]", "master", 0)
+            engine.setValue("[Auxiliary1]", "pregain", 0)
+            MixstreamPro.WIFI = true
+        }
+    }
+    if (value === 0) { return }
+}
+
+ 
+MixstreamPro.shift = false
+
+MixstreamPro.shiftButton = function(channel, control, value, status, group) {
+     
+     
+    MixstreamPro.shift = !MixstreamPro.shift;  
+}
+ 
+MixstreamPro.play1 = function(channel, control, value, status, group) {
+    let playStatus = engine.getValue("[Channel1]", "play_indicator")
+    let deck = script.deckFromGroup(group)
+
+    if (value === 0x00) {
+        if (MixstreamPro.shift) {
+            playStatus === 1 ? engine.brake(deck, true, 0.7) : engine.softStart(deck, true, 3);  
+        } else {
+            playStatus === 1 ? engine.setValue(group, "play", 0) : engine.setValue(group, "play", 1);
+        }
+    }
+}
+
+MixstreamPro.play2 = function(channel, control, value, status, group) {
+    let playStatus = engine.getValue("[Channel2]", "play_indicator")
+    let deck = script.deckFromGroup(group)
+
+    if (value === 0x00) {
+        if (MixstreamPro.shift) {
+            playStatus === 1 ? engine.brake(deck, true, 0.7) : engine.softStart(deck, true, 3);  
+        } else {
+            playStatus === 1 ? engine.setValue(group, "play", 0) : engine.setValue(group, "play", 1);
+        }
+    }
+} 
+
+ 
+MixstreamPro.play_indicator1 = function(channel, control, value, status, group) {
+    engine.setValue("[Auxiliary1]", "orientation", 2)
+}
+
+MixstreamPro.play_indicator2 = function(channel, control, value, status, group) {
+    engine.setValue("[Auxiliary1]", "orientation", 0)
+}
+
+MixstreamPro.jogWheelTicksPerRevolution = 894;
+MixstreamPro.jogSensitivity = 0.05;
+MixstreamPro.previousJogValue1 = 0
+MixstreamPro.previousJogValue2 = 0
+
+MixstreamPro.WheelTouch = function (channel, control, value, status, group) {
+    let deckNumber = script.deckFromGroup(group)
+
+if((MixstreamPro.slipenabledToggle1 === true && deckNumber === 1) || (MixstreamPro.slipenabledToggle2 === true && deckNumber === 2)){
+    if (value === 0x7F) {     
+        let alpha = 1.0/8;
+        let beta = alpha/32;
+        engine.scratchEnable(deckNumber, 8000, 33+1/3, alpha, beta);
+    } else {     
+        engine.scratchDisable(deckNumber);
+    }
+  }
+}
+
+
+MixstreamPro.JogLSB_1 = function(channel, control, value, status, group) {
+    
+    return 
+};

```

Comment:
why ignore the LSB of the jogwheels?

Besides, why are there also two identical jogwheel functions?